src/gui/transferlistsortmodel.cpp

```diff
@@ -159,19 +165,39 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
             const auto activeL = leftValue.toInt();
             const auto activeR = rightValue.toInt();
             if (activeL != activeR)
-                return activeL < activeR;
+                return activeL - activeR;
 
             const auto totalL = left.data(TransferListModel::AdditionalUnderlyingDataRole).toInt();
             const auto totalR = right.data(TransferListModel::AdditionalUnderlyingDataRole).toInt();
-            return totalL < totalR;
+            return totalL - totalR;
         }
 
     default:
         Q_ASSERT_X(false, Q_FUNC_INFO, "Missing comparsion case");
         break;
     }
 
-    return false;
+    return 0;
+}
+
+bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex &right) const
+{
+    Q_ASSERT(left.column() == right.column());
+
+    if (m_sortColumn != left.column())
+    {
+        m_subSortColumn = m_sortColumn;
+        m_sortColumn = left.column();
+    }
+
+    const int result = compare(left, right);
+    if (result != 0)
+        return result < 0;
+
+    if (m_subSortColumn != -1)
+        return compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn)) < 0;
+
+    return false; // last comparision gave equality
```

Comment:
I don't think this comment is correct.

Reply:
this statement will be only be executed if result == 0